src/cc65/datatype.c

```diff
@@ -751,6 +752,22 @@ unsigned TypeOf (const Type* T)
 
 
 
+unsigned FuncTypeOf (const Type* T)
+/* Get the code generator flag for calling the function */
+{
+    switch (GetUnderlyingTypeCode (T)) {
+
+        case T_FUNC:
+            return (((FuncDesc*) T->A.P)->Flags & FD_VARIADIC) ? 0 : CF_FIXARGC;
+
+        default:
+            Error ("Illegal function type %04lX", T->C);
+            return 0;
+    }
```

Comment:
I can't find a single-case-switch as recurring pattern in cc65. And e.g. GetFuncDesc() seems sort of similar regarding its structure and uses an if() statement. Is there a special reason for the switch() here?

Reply:
No special reason for that. It was copy & paste of the original code. Will change.